### PR TITLE
[WIP] migration of module sale_payment_method

### PR DESCRIPTION
**Pull request by fevxie**
migration sale_payment_method port 7.0 to 8.0

### CODE REVIEW

**Comment by guewen** [comment]
Thanks, good work :star:. I will wait for @pedrobaeza if he has something to change, and make a new PR myself for the last missing bits if necessary.

**Comment by yvaucher** [comment]
@fevxie You might want to setup your local git with email. It seems github is unable to recognize the author of your commit.

https://help.github.com/articles/setting-your-email-in-git/

And you can do:

```
 git commit --amend --author='Your name <your email>'
```

File: __unported__/sale_payment_method/payment_method.py [hunk 1]
```diff
@@ -19,39 +19,32 @@
...
 
-    def _default_company_id(self, cr, uid, context):
-        company_model = self.pool.get('res.company')
-        return company_model._company_default_get(cr, uid, 'payment.method',
-                                                  context=context)
+    name = fields.Char('Name', help="The name of the method on the backend")
+    journal_id = fields.Many2one(
+        'account.journal',
+        'Journal',
+        help="If a journal a selected, when a payment is recorded "
```
Comment (pedrobaeza) [inline __unported__/sale_payment_method/payment_method.py:38]: If a journal is selected

File: __unported__/sale_payment_method/sale.py [hunk 2]
```diff
@@ -41,56 +41,52 @@ def _get_order_from_line(self, cr, uid, ids, context=None):
...
-            type='boolean',
-            help="It indicates that sales order has at least one payment."),
-    }
-
-    def copy(self, cr, uid, id, default=None, context=None):
+    @api.one
+    @api.depends('amount_total', 'payment_ids.credit', 'payment_ids.debit')
+    def _get_amount(self):
+        paid_amount = 0
+        for line in self.payment_ids:
```
Comment (pedrobaeza) [inline __unported__/sale_payment_method/sale.py:48]: You can make it in one line:

```
self.amount_paid = sum(x.credit - x.debit for x in self.payment_ids)
```

File: __unported__/sale_payment_method/sale.py [hunk 3]
```diff
@@ -41,56 +41,52 @@ def _get_order_from_line(self, cr, uid, ids, context=None):
...
+        'account.move.line',
+        string='Payments Entries')
+
+    payment_method_id = fields.Many2one(
+        'payment.method',
+        'Payment Method',
+        ondelete='restrict')
+
+    residual = fields.Float(
+        compute=_get_amount,
```
Comment (pedrobaeza) [inline __unported__/sale_payment_method/sale.py:68]: Put compute method between quotes for being inheritable.

File: __unported__/sale_payment_method/sale.py [hunk 4]
```diff
@@ -41,56 +41,52 @@ def _get_order_from_line(self, cr, uid, ids, context=None):
...
+        digits_compute=dp.get_precision('Account'),
+        string='Balance',
+        store=False)
+
+    amount_paid = fields.Float(
+        compute=_get_amount,
```
Comment (pedrobaeza) [inline __unported__/sale_payment_method/sale.py:74]: Put compute method between quotes for being inheritable.

File: __unported__/sale_payment_method/sale.py [hunk 5]
```diff
@@ -41,56 +41,52 @@ def _get_order_from_line(self, cr, uid, ids, context=None):
...
+        digits_compute=dp.get_precision('Account'),
+        string='Amount Paid',
+        store=False)
+
+    payment_exists = fields.Boolean(
+        compute=_payment_exists,
```
Comment (pedrobaeza) [inline __unported__/sale_payment_method/sale.py:80]: Put compute method between quotes for being inheritable.

File: __unported__/sale_payment_method/sale.py [hunk 6]
```diff
@@ -41,56 +41,52 @@ def _get_order_from_line(self, cr, uid, ids, context=None):
...
+        string='Has automatic payment',
+        help="It indicates that sales order has at least one payment.")
+
+    @api.one
+    def copy(self, default=None):
```
Comment (pedrobaeza) [inline __unported__/sale_payment_method/sale.py:85]: You need to put `@api.model` for compatibility with old API

File: __unported__/sale_payment_method/sale.py [hunk 7]
```diff
@@ -300,5 +296,5 @@ def action_cancel(self, cr, uid, ids, context=None):
                     _('Cannot cancel this sales order!'),
```
Comment (pedrobaeza) [inline __unported__/sale_payment_method/sale.py:296]: It's very easy to convert this to new API with @api.multi and exceptions.Warning

**Comment by yvaucher** [comment]
BTW move from `__unported__` to base dir is missing.

**Comment by guewen** [comment]
I am working on these points

**Comment by guewen** [comment]
Here is my proposal which supersedes this one: https://github.com/OCA/e-commerce/pull/35

**Comment by guewen** [comment]
Hence can we close this issue?

**Comment by yvaucher** [comment]
@guewen There is just the authoring issue for @fevxie Otherwise we can close

**Comment by guewen** [comment]
> @guewen There is just the authoring issue for @fevxie Otherwise we can close

Fixed in my branch

**Pull request closed without merging**